README.md

```diff
@@ -155,38 +167,127 @@ obj->extensionMethod();
-It can also similarly extract a method from an **instance**
-into a function that always uses that instance as its `this` binding:\
-for example, `const arr = arr->(arr.slice); slice(1, 3);`.
-
 ## Real-world examples
 Only minor formatting changes have been made to the status-quo examples.
 
-<table>
-<thead>
-<tr>
-<th>Status quo
-<th>With binding
+### Node.js
+Node.js’s runtime depends on many built-in JavaScript global intrinsic objects
+that are vulnerable to mutation or prototype pollution by third-party libraries.
+When initializing a JavaScript runtime, Node.js therefore caches
+wrapped versions of every global intrinsic object (and its methods)
+in a [large `primordials` object][primordials.js].
+
+Many of the global intrinsic methods inside of the `primordials` object
+rely on the `this` binding.
+`primordials` therefore contains numerous entries that look like this:
+```js
+ArrayPrototypeConcat: uncurryThis(Array.prototype.concat),
+ArrayPrototypeCopyWithin: uncurryThis(Array.prototype.copyWithin),
+ArrayPrototypeFill: uncurryThis(Array.prototype.fill),
+ArrayPrototypeFind: uncurryThis(Array.prototype.find),
+ArrayPrototypeFindIndex: uncurryThis(Array.prototype.findIndex),
+ArrayPrototypeLastIndexOf: uncurryThis(Array.prototype.lastIndexOf),
+ArrayPrototypePop: uncurryThis(Array.prototype.pop),
+ArrayPrototypePush: uncurryThis(Array.prototype.push),
+ArrayPrototypePushApply: UncurryThisStaticApply<typeof Array.prototype.push>
+ArrayPrototypeReverse: uncurryThis(Array.prototype.reverse),
+```
+…and so on, where `uncurryThis` is `Function.bind.bind(Function.call)`
+(also called [“call-binding”][call-bind]).
+
+[call-bind]: https://npmjs.com/call-bind
 
-<tbody>
-<tr>
-<td>
+In other words, Node.js must **wrap** every `this`-sensitive global intrinsic method
+in a `this`-uncurried **wrapper function**,
+whose first argument is the method’s `this` value,
+using the `uncurryThis` helper function.
 
+The result is that code that uses these global intrinsic methods,
+like this code adapted from [node/lib/internal/v8_prof_processor.js][]:
+```js
+  // specifier is a string.
+  const file = specifier.slice(2, -4);
+```
+…
 ```js
-???
+  if (process.platform === 'darwin') {
+    tickArguments.push('--mac');
+  } else if (process.platform === 'win32') {
+    tickArguments.push('--windows');
+  }
+  tickArguments.push(...process.argv.slice(1));
 ```
-From ???.
+…must instead look like this:
+```js
+// Note: This module assumes that it runs before any third-party code.
+const {
+  ArrayPrototypePush,
+  ArrayPrototypeSlice,
+  StringPrototypeSlice,
+} = primordials;
+```
+…
+```js
+  const file = StringPrototypeSlice(specifier, 2, -4);
+```
+…
+```js
+  if (process.platform === 'darwin') {
+    ArrayPrototypePush(tickArguments, '--mac');
+  } else if (process.platform === 'win32') {
+    ArrayPrototypePush(tickArguments, '--windows');
+  }
+  ArrayPrototypePush(tickArguments,
+                     ...ArrayPrototypeSlice(process.argv, 1));
```

Comment:
We would use `ArrayPrototypePushApply` to avoid the unsafe array iteration here.
```suggestion
  ArrayPrototypePushApply(tickArguments,
                     ArrayPrototypeSlice(process.argv, 1));
```

Reply:
These lines were copied directly from <https://github.com/nodejs/node/blob/e46c680bf2b211bbd52cf959ca17ee98c7f657f5/lib/internal/v8_prof_processor.js>, which is up to date. Is the plan to change the code in that file, too?

Also, would the unsafe-array-iteration problem also apply to `tickArguments->$push(...process.argv->$arraySlice(1))`?

Reply:
`...` can't be safely used, so the `pushApply` pattern would probably still need a custom bind.

Yes, it would still apply, because `...` invokes a `Symbol.iterator` method on the operand.

Reply:
Ah, I see. So that would require a wrapper function anyway.

I’ll probably just replace this example with node/lib/internal/debugger/inspect.js. I’ll push a new commit in a bit.

Reply:
yes, that would work great

Reply:
Fixed in 04e752738c7ca80d5bae018f5dac94112530976a.

Reply:
> Is the plan to change the code in that file, too? I’d rather have the example match how Node is written right now, but I’m fine with this change too.

Addressed in https://github.com/nodejs/node/pull/40271.